tests/integration/codexclient.nim

```diff
@@ -44,6 +44,13 @@ proc upload*(client: CodexClient, contents: string): ?!Cid =
   assert response.status == "200 OK"
   Cid.init(response.body).mapFailure
 
+proc upload*(client: CodexClient, bytes: seq[byte]): ?!Cid =
+  proc toString(bytes: seq[byte]): string =
+    result = newString(bytes.len)
+    copyMem(result[0].addr, bytes[0].unsafeAddr, bytes.len)
```

Comment:
I was also thinking of just using `cast[string](bytes)` to avoid copying, but after seeing https://forum.nim-lang.org/t/6477 and https://forum.nim-lang.org/t/1356 it looks it is not a good idea... Going other way around feels safer.

See https://play.nim-lang.org/#pasty=KVKaEQFx